File: Beer2021/app/src/main/java/com/example/beer2021/MainActivity.java
package com.example.beer2021;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Spinner;
import android.widget.TextView;

import java.util.List;

public class MainActivity extends AppCompatActivity {
    private BeerExpert expert = new BeerExpert();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    
    // 버튼을 클릭하면 호출됨 
    public void findbeer(View v) {
        TextView brands = findViewById(R.id.brands); // 텍스트 부의 레퍼런스를 얻어옴\
        // 스피너의 레퍼런스를 얻어옴
        Spinner color = findViewById(R.id.color);
        // 스피너에서 선택한 항목을 얻음
        String beerType = String.valueOf(color.getSelectedItem());
        // 선택한 항목 표시
        List<String> brandList = expert.getBrands(beerType);
        StringBuilder sb = new StringBuilder();
        for(String brand:brandList) {
            sb.append(brand).append("\n");
        }
        brands.setText(sb.toString());
    }
}

File: KTFinal801/KTFinal801/app/src/main/java/com/example/ktfinal801/MainActivity.kt
package com.example.ktfinal801

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import com.example.ktfinal801.databinding.ActivityMainBinding
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {
    //지연 초기화 - var로 선언된 프로퍼티만 가능,  프로퍼티에 대한 getter/setter사용불가
    private lateinit var binding: ActivityMainBinding
    //자바스럽게 이른 인스턴스화
    private val auth: FirebaseAuth = FirebaseAuth.getInstance()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        //setContentView(R.layout.activity_main)
        binding.helloworldTextView.setOnClickListener {
            Log.i("MainActivity","click event")
            startActivity(Intent(this, LoginActivity::class.java))
        }
    }

    override fun onStart() {
        super.onStart()
        if(auth.currentUser == null){
            startActivity(Intent(this, LoginActivity::class.java))
        } else {
            startActivity(Intent(this, LikeActivity::class.java))
            finish()
        }
    }
}